.github/workflows/release.yml

```diff
@@ -0,0 +1,168 @@
+            path: target/aarch64-unknown-linux-gnu/release/jsbuilder
+            artifact_name: jsbuilder-arm-linux
+            asset_name: jsbuilder-arm-linux-${{ github.event.release.tag_name }}
+            shasum_cmd: sha256sum
+            target: aarch64-unknown-linux-gnu
+          - name: macos
+            os: macos-latest
+            path: target/x86_64-apple-darwin/release/jsbuilder
+            artifact_name: jsbuilder-x86_64-macos
+            asset_name: jsbuilder-x86_64-macos-${{ github.event.release.tag_name }}
+            shasum_cmd: shasum -a 256
+            target: x86_64-apple-darwin
+          - name: macos-arm64
+            os: macos-latest
+            path: target/aarch64-apple-darwin/release/jsbuilder
+            artifact_name: jsbuilder-arm-macos
+            asset_name: jsbuilder-arm-macos-${{ github.event.release.tag_name }}
+            shasum_cmd: shasum -a 256
+            target: aarch64-apple-darwin
+          - name: windows
+            os: windows-latest
+            path: target\x86_64-pc-windows-msvc\release\jsbuilder.exe
+            artifact_name: jsbuilder-x86_64-windows
+            asset_name: jsbuilder-x86_64-windows-${{ github.event.release.tag_name }}
+            shasum_cmd: sha256sum
+            target: x86_64-pc-windows-msvc
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # Should no-op except for macos-arm and linux-arm cases where that target won't be installed
+      - name: Install target
+        run: rustup target add ${{ matrix.target }}
+
+      # wasmtime-fiber and binaryen fail to compile without this
+      - name: Install Aarch64 GCC toolchain
+        run: sudo apt-get --assume-yes install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
+        if: matrix.target == 'aarch64-unknown-linux-gnu'
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: engine
+          path: ts/compile/builder/target/wasm32-wasi/release/
+
+      - name: Build CLI ${{ matrix.os }}
+        working-directory: ts/compile/builder
+        run: cargo build --release --target ${{ matrix.target }} --package jsbuilder
+        env:
+          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: aarch64-linux-gnu-gcc
+
+      - name: Archive assets
+        run: gzip -k -f ts/compile/builder/${{ matrix.path }} && mv ts/compile/builder/${{ matrix.path }}.gz ${{ matrix.asset_name }}.gz
+
+      - name: Copy for artifact
+        run: cp ${{ matrix.asset_name }}.gz ${{ matrix.artifact_name }}.gz
+
+      - name: Upload assets to artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.artifact_name }}.gz
+          path: ${{ matrix.artifact_name }}.gz
+
+      - name: Upload assets to release
+        if: github.event_name == 'release'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh release upload ${{ github.event.release.tag_name }} ${{ matrix.asset_name }}.gz
+
+  pull-request:
```

Comment:
We should set this action to only run on PRs if the files changed:

https://stackoverflow.com/a/70569968

The build one should always run though

Reply:
271ab8ec3ecefeb4bd485357a9e4e1bd00a0c361